sphinx_gallery/gen_rst.py

```diff
@@ -498,7 +499,8 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
+    filename_pattern = gallery_conf.get('filename_pattern', 'plot_*')
+    if not fnmatch.fnmatch(fname, filename_pattern):
```

Comment:
I can't remember why I implemented the logic like this, when incorporating the notebook styled examples. I checked back the git history. But I now I think it makes more sense,  to execute the matched file and just text parse the rest. Just flip the content of the If - else and remove the not.
If somehow the gallery breaks, put a comment to it